index.js

```diff
@@ -73,8 +75,25 @@ function getMainArgs() {
 }
 
 const protoKey = '__proto__';
-function storeOptionValue(options, longOption, value, result) {
-  const optionConfig = options[longOption] || {};
+
+function storeOptionValue(strict, options, longOption, value, result) {
+  const hasOptionConfig = ObjectHasOwn(options, longOption);
+
+  if (strict) {
+    if (!hasOptionConfig) {
+      throw new ERR_UNKNOWN_OPTION(longOption);
+    }
+
+    if (options[longOption].type === 'string' && value == null) {
+      throw new ERR_INVALID_OPTION_VALUE(`Missing value for '${longOption}' option with type:'string'`);
```

Comment:
I think will have to pass in extra parameter to include the short option the user actually used in the error message. 
(The passed context has been steadily growing! I have been wondering about making the parameters an Object so the parameters are named in the calling code, or making the whole routine an internal function to `parseArgs` so has access to locals although that smacks a bit of globals and side-affects.)

Reply:
Agreed, it's starting to get unmanageable. We can probably clean things up with a `storeOptionCreator` higher-order function that holds the static values and returns a `storeOption` function for setting the dynamic bits.

```js
function parseArgs({
  strict = true,
  options = {},
}) {
 const result = {...}

  const storeOption = storeOptionCreator({
    strict,
    options,
    result,
  });

  if (isLongOption(arg)) {
    storeOption({ longOption, optionValue });
  }

  if (isShortOption(arg)) {
    storeOption({ shortOption, optionValue });
  }
}
```

Or simply make the parameters an Object as you suggested 👍

Reply:
Went with an options bag for now and updated the error messages: https://github.com/pkgjs/parseargs/pull/74/commits/c4271a065b722e027601c94e019dc2e8a7a04434

Reply:
Note: The [error messages have been updated again](https://github.com/pkgjs/parseargs/pull/74#discussion_r847846774) based on your feedback in https://github.com/pkgjs/parseargs/pull/74#issuecomment-1095569158

Reply:
I tried the hybrid approach I had in mind (https://github.com/pkgjs/parseargs/pull/74#discussion_r846700566) and made an optional PR.